packages/headless/src/api/search/search-api-client.ts

```diff
@@ -71,6 +73,13 @@ export class SearchAPIClient {
       ...this.options,
     });
 
+    if (response instanceof Error) {
+      if (response instanceof DisconnectedError) {
+        return {error: buildDisconnectedError()};
+      }
+      throw response;
+    }
```

Comment:
Prior to returning errors, the type of `call` was only `Response`, which made Typescript think that `response.json()` always existed.

Unfortunately, with no internet connection, `call` returned an error instead, which made the call to `response.json()` below throw a different error.